### PR TITLE
Exhaustively check cases in switch statements

### PR DESCRIPTION
**Pull request by NoahTheDuke**
Supersedes https://github.com/Batterii/eslint-config-vurvey-node/pull/2 after consulting with front end team.

### CODE REVIEW

**Review by sripberger — APPROVED** [review]

**Comment by sripberger** [comment]
Since this is the first proposed change to the base rules, I'm trying to establish some process with it. Greg, Noah, and I have already discussed and agreed to this change in slack, but I want to also give Matt the opportunity to see it and raise any objections before we merge it in since I think that will be good practice.

If we'd like to add in any of our newer devs to these reviews as well, we can do that. I'm unsure if GH has a way of accomplishing this easily. Does anybody know if requesting review from a team will end up requiring approval from every member of that team? If so then that might be our best bet.

**Comment by DrunkPyromaniac** [comment]
Does anybody know if requesting review from a team will end up requiring approval from every member of that team?
> As far as I can tell, you can add separate teams to `CODEOWNERS`, then any member of that team can approve on behalf of it.  I don't think there's a way to require the approval of all members of a team.

**Review by DrunkPyromaniac — APPROVED** [review]

**Review by mdkiemle — APPROVED** [review]

**Pull request merged**
